solr/core/src/test/org/apache/solr/cli/PostToolTest.java

```diff
@@ -303,7 +303,7 @@ public void testDoFilesMode() throws IOException {
   public void testDetectingIfRecursionPossibleInFilesMode() throws IOException {
     PostTool postTool = new PostTool();
     postTool.recursive = 1; // This is the default
-    File dir = getFile("exampledocs");
+    File dir = getFile("exampledocs").toFile();
```

Comment:
Agreed. I actually had done a lot of this work already across the repo but there were so many tests using `File` and I tried to avoid the conversions or going through `File`, the changes ended up over 800+ lines... So holding those code changes on my Fork for now.

For now, I was planning on keeping the conversions of toFile() and toPath() and we can move out of all of these conversions as best as possible on another PR. Should be easy with a search. I think here I was trying to get as many of the APIs return types and parameters off of File to keep in scope so it was reviewable.